chromium_src/third_party/blink/renderer/core/dom/document.cc

```diff
@@ -18,10 +18,42 @@
 #include "third_party/blink/renderer/platform/graphics/static_bitmap_image.h"
 #include "third_party/blink/renderer/platform/graphics/unaccelerated_static_bitmap_image.h"
 #include "third_party/blink/renderer/platform/heap/handle.h"
+#include "third_party/blink/renderer/platform/network/network_utils.h"
 #include "third_party/blink/renderer/platform/supplementable.h"
 
 namespace {
 
+//  Returns the eTLD+1 for the top level frame the document is in.
+//
+//  Returns the eTLD+1 (effective registrable domain) for the top level
+//  frame that the given document is in. This includes frames that
+//  are disconnect, remote or local to the top level frame.
+//
+//  If the top level frame is not discoverable, an empty string is returned.
+//  This can happen in cases including frames that are not connected
+//  to the document, or iframes that are accessed while they're moving
+//  between processes (e.g., a frame's url is being changed to
+//  be LocalFrame to a RemoveFrame, or vise versa).
+std::string TopETLDPlusOneForDoc(const Document& doc) {
+  const auto* top_loc_frame = doc.GetFrameOfMasterDocument();
+  if (top_loc_frame == nullptr) {
+    LOG(ERROR) << "Unable to farble frame bc can't find top local frame: "
```

Comment:
should we have a DCHECK here to see if it's still happening with this change or do we still expect it to be null in some cases?